### PR TITLE
Improve test organization and coverage for `eval`

### PR DESCRIPTION
**Pull request by jugglinmike**
This work involved re-organizing (and in some cases, refactoring) existing
tests. I'm submitting re-naming and re-factoring operations as distinct commits
in order to more clearly demonstrate the provinance of each file in the new
structure.

This changeset relies on gh-570.

### CODE REVIEW

**Comment by jugglinmike** [comment]
I've just pushed two new tests to assert that local bindings created via `eval` may be deleted. These tests only apply to non-strict direct eval. This patch already had coverage for the configurability of global bindings (accounting for non-strict indirect eval), and it's not possible to verify this behavior for strict eval in any case (due to restrictions on the `delete` operator in strict mode code).

File: test/language/eval-code/indirect/var-env-var-non-strict.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-performeval
+es5id: 10.4.2-2-c-1
```
Comment (leobalter) [inline test/language/eval-code/indirect/var-env-var-non-strict.js]: are these files supposed to keep `es5id`?

Comment (jugglinmike) [inline test/language/eval-code/indirect/var-env-var-non-strict.js]: Yes. At the moment, we're still considering using that tag as a indicator that the test applies to ES5: https://github.com/tc39/test262/issues/569. As I mentioned there, I'd like to simplify that requirement, but in the mean time, I decided to just include the es5id.

**Comment by leobalter** [comment]
rebased, squashed the fixup commits and pushed to master.

**Pull request closed without merging**